model/generate.go

```diff
@@ -185,6 +181,31 @@ func (g *GeneratedProject) Save(ctx context.Context, p *Project, pp *ParserProje
 	return nil
 }
 
+// if the parser project is more recent, update contingent on that and force update the version (and vice versa)
+func updateVersionAndParserProject(v *Version, pp *ParserProject) error {
+	if pp.ConfigUpdateNumber > v.ConfigUpdateNumber {
+		curNumber := pp.ConfigUpdateNumber
+		if err := pp.UpsertWithConfigNumber(curNumber, true); err != nil {
+			return errors.Wrapf(err, "error upserting parser project '%s'", pp.Id)
+		}
+
+		if err := VersionUpdateConfig(v.Id, v.Config, curNumber, false); err != nil {
+			return errors.Wrapf(err, "database error updating version '%s'", v.Id)
+		}
+		return nil
+	}
+
+	if err := VersionUpdateConfig(v.Id, v.Config, v.ConfigUpdateNumber, true); err != nil {
+		return errors.Wrapf(err, "error updating version '%s'", v.Id)
+	}
+
+	if err := pp.UpsertWithConfigNumber(v.ConfigUpdateNumber, false); err != nil {
+		return errors.Wrapf(err, "database error upserting parser project '%s'", pp.Id)
+	}
+	v.ConfigUpdateNumber += 1
```

Comment:
nit: Any reason this can't be in `VersionUpdateConfig` for consistency?

Reply:
Struggled with this a little. Mostly I'm not sure because in the other case we aren't guaranteed that version was actually updated. The only reason we even need this field to be updated in memory is for test cases, so maybe I'll just have those tests re-look up versions?